### DISCUSSION TITLE
Access the broker on Kubernetess without using pulsar-proxy

### DISCUSSION THREAD

**Discussion by thomasechen** [apache/pulsar#20854] [Q&A]
We setup a pulsar cluster on Kubernetess successfully, We would like to expose the broker service to the clients outside the kubernetess.
Considering the cost of traffic using pulsar-proxy. We decided to expose the broker service directly without using pulsar proxy.

For the broker service, which way do you recommend to use ? NodePort or Loadbalancer

**Comment by ericsyh** [comment]
> We decided to expose the broker service directly without using pulsar proxy.

Normally speaking, exposing the broker service directly with NodePort or Loadbalancer can't work cause during the topic lookup process the broker will return the k8s internal DNS address to the client which is not reachable. And that's a design goal for Pulsar Proxy to help on the topic lookup under this scenario. 

If you really want to terminate the Pulsar Proxy, you mean need another proxy supports SNI Routing https://pulsar.apache.org/docs/3.0.x/concepts-proxy-sni-routing/ one popular is Istio. 
There is an example how to use the Istio to expose Pulsar https://docs.streamnative.io/platform/istio but it has dependency of 
* Using https://github.com/streamnative/charts/tree/master/charts/pulsar-operator because that operator helps manage Istio resources 
* Support wildcard DNS name in your environment.

**Comment by lhotari** [comment]
Related Apache Pulsar Helm chart issue: https://github.com/apache/pulsar-helm-chart/issues/423
